src/main/java/org/mockito/internal/util/reflection/FieldInitializer.java

```diff
@@ -284,9 +279,17 @@ private int countMockableParams(Constructor<?> constructor) {
 
         @Override
         public FieldInitializationReport instantiate() {
+            Field[] fields = testClass.getClass().getDeclaredFields();
+            HashMap<String, Type> hashmap = new HashMap<>();
+
+            for (Field field : fields) {
+                hashmap.put(field.getName(), field.getGenericType());
```

Comment:
Do we really need this hashmap? You are comparing objects with strings in the hashmap in the implementation, but that can lead to false-positives. I am also not sure why we would do that lookup there, as we can also pass in all the fields of the testclass into the lookup and perform the injection there.

Reply:
Hi thanks for the feedback! Definitely do agree that the hashmap is quite redundant but there's an issue storing mocks as Objects, as this causes type erasure. I'll try to remove this hashmap in the next commit as well as storing mocks as type Type instead of Object.